### PR TITLE
Fixed light gbm update

### LINKED ISSUE #7045: LightGBM Version
**System Information (please complete the following information):**
 - OS & Version: Windows 11 Pro 23H2 build 22631.3235
 - ML.NET Version: ML.NET version 3.0.1
 - .NET Version: .NET 8.0

**Describe the bug**

System.AccessViolationException
HResult=0x80004003
Message=Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

**To Reproduce**
1. Make a console app project
2. Install following packages with NuGet Package Manager:
Microsoft.ML Version 3.0.1
Microsoft.ML.AutoML Version 0.21.1
Microsoft.ML.CpuMath Version 3.0.1
Microsoft.ML.DataView Version 3.0.1
Microsoft.ML.FastTree Version 3.0.1
Microsoft.ML.LightGbm Version 3.0.1
Microsoft.ML.Mkl.Compreonents Version 3.0.1
Microsoft.ML.Mkl.Redist Version 3.0.1
3. Edit Program.cs:
```
using Microsoft.ML;
using Microsoft.ML.AutoML;
using Microsoft.ML.Data;
using static Microsoft.ML.DataOperationsCatalog;

MLContext mlContext;
string sampleData = "taxi-fare-full.csv";
mlContext = new MLContext();

// Infer column information
ColumnInferenceResults columnInference = mlContext.Auto()
                                                .InferColumns(sampleData,
                                                    labelColumnName: "fare_amount",
                                                    groupColumns: false);
// Create text loader
TextLoader loader = mlContext.Data.CreateTextLoader(columnInference.TextLoaderOptions);

// Load data into IDataView
IDataView data = loader.Load(sampleData);

TrainTestData trainValidationData = mlContext.Data.TrainTestSplit(data, testFraction: 0.2);

SweepablePipeline pipeline = mlContext.Auto()
                    .Featurizer(data, columnInformation: columnInference.ColumnInformation)
                        .Append(mlContext.Auto()
                            .Regression(labelColumnName: columnInference.ColumnInformation.LabelColumnName));

AutoMLExperiment experiment = mlContext.Auto().CreateExperiment();

var regressionMetric = RegressionMetric.RootMeanSquaredError;
experiment
    .SetPipeline(pipeline)
    .SetRegressionMetric(regressionMetric,
                            labelColumn: columnInference.ColumnInformation.LabelColumnName)
    .SetTrainingTimeInSeconds(100)  // Training time in sec
    .SetDataset(trainValidationData);

// Log experiment trials
mlContext.Log += (_, e) => {
    if (e.Source.Equals("AutoMLExperiment"))
    {
        Console.WriteLine(e.RawMessage);
    }
};

TrialResult experimentResults = await experiment.RunAsync();
```
4.This works without  errors. Add LightGBM version 4.0.0 or latest 4.3.0 with NuGet Package Manager.
5. See error

**Additional context**
Without installing LightGBM version 4.?.?, ML.NET version 3.0.1 uses LightGBM version 3.3.5.
I've asked [Version 3.3.5 to Version 4.3.0](https://github.com/microsoft/LightGBM/issues/6309) at microsoft/LightGBM.
LightGBM Version 4.3.0 itself maybe works fine. So it might be the problem how ML.NET calls LightGBM.

### LINKED ISSUE #7320: Update LightGBM to 4.5.0
**System Information (please complete the following information):**
 - OS & Version: Windows 11 Pro
 - ML.NET Version: v4.0.0
 - .NET Version: .NET9

**Describe the bug**
Since ML.NET 4.0 uses LightGBM 3.3.5, I install LightGBM v4.5.0 by NuGet, but still having a problem with that version.

**To Reproduce**
1. Install ML.NET by NuGet
2. Install LightGBM v4.5.0
3. Try run
4. Image

It has been a problem using LightGBM v4.5.0 with ML.NET.  Releasing v4.0,  I hoped to go better with LightGBM v4.5.0, well, no progress has been done, I guess.  Is this something that you don't care about?  Well, I think I've already reported this problem but not getting any efort.

I wish someone check this problem and if those of you working on ML.NET sticks with LightGBM v3.3.5 explane why LihgtGBM v4.5.0 won't be supported with the latest version of ML.NET.

### PR DESCRIPTION
**Pull request by michaelgsharp**
Fixes #7320.
Fixes #7045.
Updated light GBM to the latest version.

### CODE REVIEW

**Comment by LittleLittleCloud** [comment]
Can LightGBM 4.* load model created by LightGBM 3.*?

**Comment by ericstj** [comment]
> Can LightGBM 4.* load model created by LightGBM 3.*?

I saw mention of this here https://github.com/microsoft/LightGBM/issues/2228#issuecomment-501923283  it seems that what we are doing should be supported.  I was hoping to find something about it in the [docs](https://lightgbm.readthedocs.io/en/stable/) though.

@michaelgsharp - oddly it seems like TensorFlow tests are failing on windows jobs in this PR with higher accuracy.  I'm not sure what to think about that - but it's pretty consistent across multiple machines.  🤔

**Comment by michaelgsharp** [comment]
@ericstj just looked into the failing test. It looks like in that case TensorFlow is used only to _featurize_ the data. That data is then fed into a LightGBM pipeline.. so thats the corrolation between them.

**Comment by michaelgsharp** [comment]
I'll get the test updated with the new values since it does seem to be directly related to lightGBM in this specific case.

**Comment by ericstj** [comment]
So it looks like this LightGBM update improved the accuracy of that test.  Bonus!

**Review by ericstj — APPROVED** [review]
LGTM

**Comment by ericstj** [comment]
Failures are known due to file sharing between tests @LittleLittleCloud
